### DISCUSSION TITLE
Explore possibilities for newer and better Atmos System

### DISCUSSION THREAD

**Discussion by krille90** [unitystation/unitystation#7541] [General]
## Description

The current Atmos system is still very fragile and can break easily due to concurrency issues.
The idea is to eventually move Atmos to a ECS and/or c# Jobs approach.

This ticket is to keep track of the discussion.

**Comment by krille90** [comment]
__Idea 1__
* Each job gets a chunk of tiles, e.g. 16x16 tiles (at least 2x2)
* Atmos will be updated in four steps to avoid updating happening at the same time:
 
    ![image](https://user-images.githubusercontent.com/7312905/74198971-238ae580-4c63-11ea-91cf-82645ab15fac.png)

* TODO: Handling exchange between two matrices

**Comment by krille90** [comment]
__Idea 2__
* two buffer system
* like a waterflow simulation
* gas flow only from high to low pressure
* TODO: more details to follow ...

**Comment by RedlineTriad** [comment]
I have thought about this before, this is what i came up with without including combustion.
The issue i see is that every gas is everywhere, this is great for performance but would make something like atmospheric chemistry more memory expensive.
![image](https://user-images.githubusercontent.com/39059512/74366484-67e7c400-4dd0-11ea-9de5-9c8d44b18df5.png)

**Comment by Macoron** [comment]
Posted this before in atmos thread:
https://docs.google.com/document/d/18gcuNQO_E3vciqY7uPzt_VAS_S0afK3QOSsJg9GBB7Q/edit?usp=sharing

Just put my thoughts on a paper, open for feedback.

**Comment by PerfectTangent** [comment]
converting to discussion.
